.github/workflows/prometheus-rules.yml

```diff
@@ -6,18 +6,18 @@ on:
     paths:
     - observability/prometheus/rules/**/*.y*ml
 
-env:
-  GO_VERSION: ~1.21
-
 jobs:
   rules:
     runs-on: ubuntu-latest
     steps:
+      - uses: actions/checkout@v4
+        name: Checkout code
+
       - name: Install Go
         uses: actions/setup-go@v5
         with:
-          go-version: ${{ env.GO_VERSION }}
-      - uses: actions/checkout@v4
+          go-version-file: "go.mod"
```

Comment:
Nice, I didn't know this was an option, definitely going to grab this for other places